TCAT/Controllers/RouteOptionsViewController.swift

```diff
@@ -320,7 +321,65 @@ class RouteOptionsViewController: UIViewController {
+        // Reload table every time update timer is fired
+        routeResults.reloadData()
+    }
+    
+    private func getDelay(tripId: String, stopId: String) -> Future<Response<Int?>> {
+        return networking(Endpoint.getDelay(tripID: tripId, stopID: stopId)).decode()
+    }
+
+    @objc func updateLiveTracking(sender: Timer) {
+        for routesArray in routes {
+            for (index, route) in routesArray.enumerated() {
+                if !route.isRawWalkingRoute(),
+                    let direction = route.getFirstDepartRawDirection(),
+                    let tripId = direction.tripIdentifiers?.first,
+                    let stopId = direction.stops.first?.id {
+                    getDelay(tripId: tripId, stopId: stopId).observe(with: { result in
+                        let fileName = "RouteTableViewCell"
+                        DispatchQueue.main.async {
+                            switch result {
+                            case .value (let delayResponse):
+                                guard (delayResponse.data != nil), let delay = delayResponse.data else {
+                                    return
+                                }
+                                // LUCY - Check
+                                let isNewDelayValue = (route.getFirstDepartRawDirection()?.delay != delay)
+                                if isNewDelayValue {
+                                    JSONFileManager.shared.logDelayParemeters(timestamp: Date(), stopId: stopId, tripId: tripId)
+                                    JSONFileManager.shared.logURL(timestamp: Date(), urlName: "Delay requestUrl", url: Endpoint.getDelayUrl(tripId: tripId, stopId: stopId))
+                                    if let data = try? JSONEncoder().encode(delayResponse) {
+                                        do { try JSONFileManager.shared.saveJSON(JSON.init(data: data), type: .delayJSON(rowNum: index)) } catch let error {
+                                            let line = "\(fileName) \(#function): \(error.localizedDescription)"
+                                            print(line)
+                                        }
+                                    }
+                                }
+                                let departTime = direction.startTime
+                                let delayedDepartTime = departTime.addingTimeInterval(TimeInterval(delay))
+                                let isLateDelay = (Time.compare(date1: delayedDepartTime, date2: departTime) == .orderedDescending)
+                                if isLateDelay {
+                                    let delayState = DelayState.late(date: delayedDepartTime)
+                                    self.delayDictionary[route.routeId] = delayState
+                                } else {
+                                    let delayState = DelayState.onTime(date: departTime)
+                                    self.delayDictionary[route.routeId] = delayState
```

Comment:
can't this line be moved out of the `if / else` since it occurs in both cases?

this logic also seems like it could be simplified, do we need to keep track of both `departTime` and the `delayedDepartTime`, or do we only need to keep track of one?

another way to clean this up might be to have `DelayState` take a `time` and an `enum` representing the state, so that we don't have to call two separate constructors for the same use case!